database/ticket.go

```diff
@@ -142,6 +144,13 @@ func (vdb *VspDatabase) DeleteTicket(ticket Ticket) error {
 }
 
 func (vdb *VspDatabase) UpdateTicket(ticket Ticket) error {
+	// Marshal ticket first. This could prevent a DB transaction being started
+	// and rolled back unnecessarily.
+	ticketBytes, err := json.Marshal(ticket)
```

Comment:
This would actually have the function do more work than needed if the ticket does not exist in the db. Having the typ marshalled on just before persisting is better I think.